integration_test/main.go

```diff
@@ -0,0 +1,167 @@
+type Config struct {
+	config_name string
+	proxy_url string
+
+	// the headers that are expected to be included/excluded in detection
+	header_include []string
+	header_exclude []string
+
+	// true if the config for cookie contains `include`
+	// in this case, only the cookies specified in cookie_included will
+	// be included in detection
+	cookie_include_bool bool
+
+	// the cookie names that are expected to be included/excluded in detection
+	cookie_include []string
+	cookie_exclude []string
+
+	// true if the config for body contains `include`
+	body_include_bool bool
+
+	// the body param keys that are expected to be included/excluded in detection
+	body_include []string
+	body_exclude []string
+}
+
+func TestForConfig(config Config) {
+	suite := NewTestSuite(config.proxy_url)
+	sqli_string := "=1' AND 1=1"
+	var request TestRequest
+
+	// no SQL injection
+	suite.requests["Request without SQL injection"] = NewTestRequest()
+
+	// path
+	request = NewTestRequest()
+	request.pathParams.Set("path-key-0", sqli_string)
+	request.sqli = true
+	suite.requests["Request with SQL injection in path value"] = request
+
+	request = NewTestRequest()
+	request.pathParams.Set(sqli_string, "path-val-0")
+	request.sqli = true
+	suite.requests["Request with SQL injection in path key"] = request
+
+	// header
+	for _, key := range config.header_include {
+		request = NewTestRequest()
+		request.headers[key] = sqli_string
+		request.sqli = true
+		suite.requests["Request with SQL injection in included header: " + key] = request
+	}
+	for _, key := range config.header_exclude {
+		request = NewTestRequest()
+		request.headers[key] = sqli_string
+		request.sqli = false
+		suite.requests["Request with SQL injection in excluded header: " + key] = request
+	}
+
+	// cookie
+	for _, key := range config.cookie_include {
+		request = NewTestRequest()
+		request.cookies[key] = sqli_string
+		request.sqli = true
+		suite.requests["Request with SQL injection in included cookie: " + key] = request
+	}
+	for _, key := range config.cookie_exclude {
+		request = NewTestRequest()
+		request.cookies[key] = sqli_string
+		request.sqli = false
+		suite.requests["Request with SQL injection in excluded cookie: " + key] = request
+	}
+	if config.cookie_include_bool == false {
+		request = NewTestRequest()
+		request.cookies[sqli_string] = "value"
+		request.sqli = true
+		suite.requests["Request with SQL injection in cookie key"] = request
+	}
+
+	// body
+	for _, key := range config.body_include {
+		request = NewTestRequest()
+		request.body.Set(key, sqli_string)
+		request.sqli = true
+		request.sqli_body = true
+		suite.requests["Request with SQL injection in included body param: " + key] = request
+	}
+	for _, key := range config.body_exclude {
+		request = NewTestRequest()
+		request.body.Set(key, sqli_string)
+		request.sqli = false
+		suite.requests["Request with SQL injection in excluded body param: " + key] = request
+	}
+	if config.body_include_bool == false {
```

Comment:
should omit comparison to bool constant, can be simplified to `!config.body_include_bool`

Reply:
Updated in this [commit](https://github.com/dengyijia/WAF-wasm/pull/3/commits/0366c98d67882ade5042e11e83d1095fab624aae)